internal/generator/vector/outputs.go

```diff
@@ -23,6 +23,7 @@ func Outputs(clspec *logging.ClusterLoggingSpec, secrets map[string]*corev1.Secr
 			outputs = generator.MergeElements(outputs, loki.Conf(o, inputs, secret, op))
 		case logging.OutputTypeElasticsearch:
 			outputs = generator.MergeElements(outputs, elasticsearch.Conf(o, inputs, secret, op))
+			op[logging.OutputTypeElasticsearch] = true
```

Comment:
not clear why we are adding `Elasticsearch` flag to a function which produces Elasticsearch config

Reply:
> The solution seems to work, but not obvious why

@vimalk78: I am just keeping a flag `op[logging.OutputTypeElasticsearch] = true` which remembers that pre_process element is already generated and using this flag  in elasticsearch.Conf method to generate pre_process element only once.

Reply:
As discussed this has been fixed with sol-1 suggestion